akka-stream/src/main/scala/akka/stream/javadsl/FlowWithContext.scala

```diff
@@ -249,5 +243,6 @@ final class FlowWithContext[-In, -CtxIn, +Out, +CtxOut, +Mat](
         Out2,
         CtxOut2,
         Mat2]): FlowWithContext[In2, CtxIn2, Out2, CtxOut2, Mat2] =
-    new FlowWithContext(f(delegate))
+    f(this.asScala).asJava
```

Comment:
Hmm, kind of unfortunate switching between java and scala API's has such a runtime cost like this... hard to prevent though I guess, the alternative being the Scala API never has the cost and any Java implementations always do...

Reply:
Yeah, or duplicate the actual impl